### PR TITLE
Testing/bookmarks

### PR DESCRIPTION
**Pull request by kspeer825**
# Description of change
Upgrade bookmarks test to ensure replication key usage and record counts.

# Manual QA steps
 - None
 
# Risks
 - None, test additions. 
 
# Rollback steps
 - revert this branch

### CODE REVIEW

File: tests/test_facebook_bookmarks.py [hunk 1]
```diff
@@ -43,73 +45,224 @@ def expected_sync_streams():
...
+        self.end_date = self.get_properties()['end_date']
+        self.bookmarks_test(insight_streams)
+
+        # Testing agianst core objects
+        self.end_date = '2021-02-09T00:00:00Z'
+        self.bookmarks_test(non_insight_streams)
+
+
+    def bookmarks_test(self, expected_streams):
+        """A Parametrized Bookmakrs Test"""
```
Comment (luandy64) [inline tests/test_facebook_bookmarks.py:117]:
```suggestion
        """A Parametrized Bookmarks Test"""
```

File: tests/test_facebook_bookmarks.py [hunk 2]
```diff
@@ -43,73 +45,224 @@ def expected_sync_streams():
...
+        first_sync_bookmarks = menagerie.get_state(conn_id)
+
+        ##########################################################################
+        ### Update State Between Syncs
+        ##########################################################################
+
+        new_states = {'bookmarks': dict()}
+        simulated_states = self.calculated_states_by_stream(first_sync_bookmarks)
+        for stream, new_state in simulated_states.items():
+            replication_key = list(expected_replication_keys[stream])[0]
```
Comment (luandy64) [inline tests/test_facebook_bookmarks.py:147]: Couldn't this be avoided if `calculated_states_by_stream` hands back the new state? It already knows what the replication key for each stream is, right?

File: tests/test_facebook_bookmarks.py [hunk 3]
```diff
@@ -43,73 +45,224 @@ def expected_sync_streams():
...
+                first_bookmark_key_value = first_sync_bookmarks.get('bookmarks', {stream: None}).get(stream)
+                second_bookmark_key_value = second_sync_bookmarks.get('bookmarks', {stream: None}).get(stream)
+
+
+                if expected_replication_method == self.INCREMENTAL:
+
+
+                    # collect information specific to incremental streams from syncs 1 & 2
+                    replication_key = list(expected_replication_keys[stream])[0]
+                    self.assertEqual(1, len(list(expected_replication_keys[stream]))) # catches unexpected compound replication key
```
Comment (luandy64) [inline tests/test_facebook_bookmarks.py:188]: Why do we cast to `list` first? Isn't `expected_replication_keys[stream]` a set?

**Review by luandy64 — APPROVED** [review]

File: tests/test_facebook_bookmarks.py (hunk 3 above)
Comment (kspeer825) [reply tests/test_facebook_bookmarks.py:188]: Just wanted to grab the first item. Could do next(iter(expected_replication_keys[stream])) instead.

File: tests/test_facebook_bookmarks.py (hunk 2 above)
Comment (kspeer825) [reply tests/test_facebook_bookmarks.py:147]: Yeah, I can do that. It makes more sense your way.

File: tests/test_facebook_bookmarks.py (hunk 1 above)
Comment (kspeer825) [reply tests/test_facebook_bookmarks.py]: Done

File: tests/test_facebook_bookmarks.py (hunk 3 above)
Comment (kspeer825) [reply tests/test_facebook_bookmarks.py]: Done.

File: tests/test_facebook_bookmarks.py (hunk 2 above)
Comment (kspeer825) [reply tests/test_facebook_bookmarks.py]: Done.

**Pull request merged**